brage-import/src/main/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumer.java

```diff
@@ -279,13 +284,24 @@ private UriWrapper discardedFilesReportUri(Publication publication, S3Event s3Ev
     private BrageMergingReport persistInDatabaseAndCreateMergeReport(Publication publicationForUpdate,
                                                                      Publication existinPublication,
                                                                      PublicationRepresentation representation) {
+
         var importSource = createImportSource(representation);
         var resource = Resource.fromPublication(publicationForUpdate);
+        var updatedFiles = resource.getFiles();
+        var oldFiles = Resource.fromPublication(existinPublication).getFiles().stream().map(File::getIdentifier).toList();
+        updatedFiles.stream()
+            .filter(file -> containsNotFile(file, oldFiles))
+            .forEach(file -> FileEntry.importFileEntry(file, resource.getIdentifier(),
+                                              UserInstance.fromPublication(representation.publication()), importSource).persist(resourceService));
         resource.updateResourceFromImport(resourceService, importSource);
         var newImage = resource.fetch(resourceService).orElseThrow().toPublication();
         return new BrageMergingReport(existinPublication, newImage);
     }
 
+    private static boolean containsNotFile(File file, List<UUID> fileList) {
```

Comment:
how about removing the "!" and wrapping the call in Predicate::not?

Reply:
Nice, have never seen it